### DISCUSSION TITLE
My Nova Core Wish List

### DISCUSSION THREAD

**Discussion by stuartcusackie** [laravel/nova-issues#3417] [Ideas & Feature Requests]
I have been dabbling in Nova for a few years now and it is great when the project is particularly suited to it, but certain use cases feel a bit hacky. I find myself using lots of third party packages, and I worry the support for each of them won't be there long term. I wish a few of these common admin features were part of the core. 

I have also started using Statamic 3, and I think Nova could borrow a few ideas from it. I know Nova is not aiming to be a CMS, and it excels in areas that Statamic cannot, but almost every single client that I build a Nova app for wants to manage some content on their front-end.

Here is my current wishlist. I'm sure I've forgotton some things so I may update this later:

- **Media library support & multi-file fields**: Statamic has integrated very nicely into Laravel's filesystems, and its easy to set up various asset containers and connect them to fields. The existing nova packages are good but I think such a key feature of Laravel should be part of the Nova core.
- **A better editor**: Trix is extremely difficult to customise, and judging by the Github issues, most Nova developers have given up on it. TipTap is much more developer friendly and there is a nova package for it, but again Statamic is miles ahead with its Bard field (also TipTap).
- **Conditional fields**: Every Nova project I have built has had a conditional fields requirement. Again the nova dependency container package is good but I think this should be part of the core.
- **Repeating fields**: As above, plenty of packages available but perhaps this could be a core feature?
- **Store multiple fields in a single JSON column**: If all nova fields could be designated to be stored in a single json column along with other fields, then it would negate the need to create migrations for every single column. I think this would be especially useful for meta data, page content and fields that get changed a lot over time. **EDIT** - I've just realised that this may be possible already. I'll check it out and update.
- **Resource Templates**: A single nova resource could have multiple templates, with different fields depending on the chosen template (similar to Statamic Blueprints). If this was available, in conjuction with JSON field storage, then we could build single-use resources that have unique fields, and store them in a single database table.

Thanks for reading!

**Comment by kirkbushell** [comment]
Adding to this:

* MOBILE SUPPORT!!!! Nova is just awful to use on mobile currently
* Ability to link any field to the details page. I do this on all my projects and never realised how useful it is until I saw Nova didn't do it.
* Back links from previous page (particularly useful from detail/edit pages)
* Saved search!!! So annoying going back and having to re-enter search every time (again, another thing I do on all my projects that I thought was default behaviour these days lol)
